source/models/UserLog.ts

```diff
@@ -5,13 +5,14 @@ import { UUID } from '../sequelizeSetup.js';
 
 export interface Log {
 	// TODO: keys for db logs
-	action?: 'GET' | 'PUT' | 'DELETE' | 'POST';
+	action?: string;
```

Comment:
why type widening?

Reply:
`req.method` is string, hence it is better to widen the DB entry to string than to write a switch statement that converts `req.method` to the type.
Also kind of future-proofs the logging middleware, since not all HTTP requests are GET, PUT, DELETE or POST (i.e. you could have OPTIONS if there's CORS involved).